src/widgets/index/index.ts

```diff
@@ -590,6 +605,13 @@ const index = (widgetParams: IndexWidgetParams): IndexWidget => {
           instantSearchInstance.onInternalStateChange();
         }
       });
+
+      if (indexInitialResults) {
+        // If there are initial results, we're not notified of the next results
+        // because we don't trigger an initial search. We therefore need to directly
+        // schedule a render that will render the results injected on the helper.
```

Comment:
This would be the code change required. I'm not sure this is worth it tbh, it's more error prone.

```diff
  if (indexInitialResults) {
    const results = new algoliasearchHelper.SearchResults(
      new algoliasearchHelper.SearchParameters(indexInitialResults._state),
      indexInitialResults._rawResults
    );

    derivedHelper.lastResults = results;
-   helper.lastResults = results;
  }

  // ...

  if (indexInitialResults) {
-   instantSearchInstance.scheduleRender();
+   derivedHelper.emit('result', {
+     results: derivedHelper.lastResults,
+     state: helper.state,
+   });
  }
```

Reply:
Why do you think it's more error-prone? Since we'd be using the helper API, we don't have to be concerned with reproducing what it does under the hood.

Reply:
We never emit the "result" event in InstantSearch, the Helper does internally. So that would be a new concept in the codebase – a new API we need to grasp. And we need to give it the correct `results` and the correct `state` (the distinction of helper and derived helper is already confusing in `index`).

On the other hand, we created `scheduleRender` in InstantSearch and already use it in `InstantSearch` and `index`.

Reply:
Makes sense 👍